README.md

```diff
@@ -156,7 +156,27 @@ class CreateUser
 end
 ```
 
-You'll get an `ArgumentError` if you pass any params not defined in the `inputs_for` block, or if you don't supply all required params. This is completely opt-in so if you don't provide an `inputs_for` block, no checking is performed.
+### Specifying types and nullable fields
+
+You can now also optionally specify types and nullability for your inputs. For the `type:` option, any ruby class can be used.
+
+```ruby
+inputs_for(:call) do
+  required :name, type: String, null: false
```

Comment:
That was a big discussion when adding the null check. As you so astutely pointed out, it doesn't really make sense to mark a required field as nullable, and it equally doesn't make sense for the default to stay nullable if you don't specify anything. One could reasonably assume that we should forbid specifying null at all on required fields and always make them required.

Although this seems reasonable, it does assume the consumer of this library has the same assumptions that I do, and that's not something I want to do. This is an instance where I explicitly reject the notion of "convention over configuration", because what's conventional to me is not necessarily conventional to someone else. Instead, I prefer the notion of "explicit is better than implicit" because being explicit forces you to make things more obvious.

The rule is "arguments can be nil". You can _explicitly_ override this rule by declaring `null: false`, at which point it's 100% obvious that the input can't be `nil`. There's no holding the extra information in your head of "arguments can be nil, _except_ if they're marked as required and then they can't be nil". I can even go a step farther and conjure up a new `required_conditionally` or `optional_conditionally` that have complex rules around when they're required/optional and when they're nullable/not nullable by default and how that default changes if/when they become conditionally required/optional. Instead of adding all that incidental complexity, I've chosen to have a simple rule and a simple way to explicitly override that rule.

As to your question of the point of required: marking a variable as `required` means you're forcing the caller of a class to know about the input and any constraints you want to place on that input.